dvc/remote/ssh/__init__.py

```diff
@@ -264,7 +264,9 @@ def open(self, path_info, mode="r", encoding=None):
 
     def list_cache_paths(self):
         with self.ssh(self.path_info) as ssh:
-            return list(ssh.walk_files(self.path_info.path))
+            # If we simply return an iterator then with above closes instantly
+            for path in ssh.walk_files(self.path_info.path):
+                yield path
```

Comment:
Wouldn't that mean that connection will be open until the iterator is closed? 
Considering that now it is used  only in `remote.all`, which was supposed to create a generator in the first place this change is good.

I am just wondering if it will not become fragile if, in the future, we will start to wrap some logic around this method (for any reason). But also, I am worrying prematurely here. I just wanted to ask this and keep in mind for the future.

Reply:
Yes, this will hold the connection open. Not sure this will be an issue.